hls4ml/optimization/dsp_aware_pruning/keras/__init__.py

```diff
@@ -15,7 +12,6 @@
 from hls4ml.optimization.dsp_aware_pruning.keras.reduction import reduce_model
 from hls4ml.optimization.dsp_aware_pruning.scheduler import OptimizationScheduler
 
-np_config.enable_numpy_behavior()
```

Comment:
Can you provide more details as to why this breaks the flow on some setups?

Reply:
Some indexing and other behaviors are changed. e.g.: in default mode `tf.reduce_all(a>b)` is fine, but will raise if this is set. This causes some qkeras quantizers to return dicts for unknown reason, and I did not track down the direct cause.
Non-comprehensive list of failed tests:
hgq, conv1d, conv1dpointwise, qkeras po2, repack stream, qkeras mnist cnn, binary cnn

Reply:
Where does this break occur? Is it with a newer version of NumPy or TensorFlow? Because the CI/CD on GitHub doesn't shows this, so I am just wondering under what environment this occurs.

Reply:
Gitlab CI test is not affected because this line is not executed before those (qkeras po2 and hgq) tests: only dsp aware pruning related tests will import the optimization api and trigger this line, and those don't run before the broken tests.

Reply:
If some tests are run with numpy behavior enabled. Full failure pattern: https://gitlab.cern.ch/fastmachinelearning/hls4ml/-/pipelines/8406093

Reply:
I see. I have 2 comments on this but in general, I am okay with merging this PR as long as the loss is actually printed with `tf.print`. First, the error causing seems a bit artificial - is it ever likely that a user is going to import both `dsp_aware_pruning` and `hgq` / `po2` in the same module. Secondly, looking at the test logs this seems like this is a TensorFlow issue in some recent update? Looking at the docs for this function: https://www.tensorflow.org/api_docs/python/tf/experimental/numpy/experimental_enable_numpy_behavior maybe the parameter `dtype_conversion_mode` can help solve this issue without changing the print statements.

Reply:
Response to your comments:
1. I don't think the errors are artificial -- if one runs the tests on one node, this is what to expect. I spend at least half an hour just to debug why some irrelevant tests breaks after some editing, and just to find out that the test execution order changed and things breaks because of this.
2. The issue is that global behavior is silently overridden by accessing a submodule of a package, and this should be avoided to the max extent possible. In this specific case, it only breaks hgq/po2 conversion hls4ml, but it could introduce undefined behavior on other places. I would suggest changing the print statement using `tf.print` to avoid affecting things outside, or pack it into a context manager to restore the original state after exiting the module.

Regarding if losses is actually printed, I don't have the setup to validate it on my side.